Source/Tools/FEXLoader/LinuxSyscalls/Syscalls.cpp

```diff
@@ -597,6 +585,12 @@ uint64_t CloneHandler(FEXCore::Core::CpuStateFrame *Frame, FEX::HLE::clone3_args
 
       args->SignalMask = ~0ULL;
       ::syscall(SYS_rt_sigprocmask, SIG_SETMASK, &args->SignalMask, &args->SignalMask, sizeof(args->SignalMask));
+
+      // In the case of thread, we need a new stack
+      // Needs to be allocated before the locks occur.
```

Comment:
Why does it hang otherwise?

Reply:
Context fork locking locks the allocator mutex, so no allocations can occur inside of that.

Reply:
I'm having trouble finding which locks are interfering here. I only looked into LockBeforeFork, which seems to lock a different mutex than AllocateStackObject.

Reply:
Lock Side:
- Thread->CTX->LockBeforeFork(Frame->Thread);
- Allocator::LockBeforeFork(Thread);
- Alloc64->LockBeforeFork(Thread);
- AllocationMutex.lock();

Stack Allocation side:
- FEX::LinuxEmulation::Threads::AllocateStackObject();
- FEXCore::Allocator::mmap(nullptr, FEX::LinuxEmulation::Threads::STACK_SIZE, PROT_READ | PROT_WRITE, MAP_PRIVATE | MAP_ANONYMOUS, -1, 0);
- FEX_DEFAULT_VISIBILITY extern MMAP_Hook mmap;
- FEXCore::Allocator::mmap = FEX_mmap;
- void *Result = Alloc64->Mmap(addr, length, prot, flags, fd, offset);
- auto lk = FEXCore::GuardSignalDeferringSectionWithFallback(AllocationMutex, TLSThread);

Reply:
Ah, I was looking at the wrong LockBeforeFork before.

This should be a bit clearer:
```suggestion
      // LockBeforeFork grabs the allocator mutex to block allocations temporarily, so this must be allocated before
```